### ISSUE TITLE
Provide a component lib boilerplate

### ISSUE THREAD

**Issue by trickstival** [vuejs/vue-cli#4066]
### What problem does this feature solve?
One of the most amazing things about vue is that you can just plug and play with third-party open-source components by installing them on your project and importing them.

Personally, when I start to develop an open-source component I can do it in two ways:

- If the component is very simple, I configure a simple bundler such as `rollup` and build it using render functions
- If it's more complex and requires css and more stuff I just build a new project with `vue create` and adjust the file structure to match my needs (to be imported)

I would really like it if there was a way to generate a slim file structure to build open-source standalone vue components. 

Maybe something like this:
`vue create --lib my_open-source_component_name`

Then it would generate a vue project with just a component and ready-to-use npm publish / versioning, vue test utils and all stuff needed to create it

If this idea provides a standardized API, maybe it would be possible to provide a "component store" on Vue UI, where you can pick a component checking examples and usage (but now I'm just daydreaming)

### What does the proposed API look like?
`vue create --lib my_open-source_component_name` generates a minimal file structure to build standalone components

**Comment by sodatea** [comment]
Is this what you need? https://twitter.com/chrisvfritz/status/1060208177852104705

**Comment by trickstival** [comment]
Yes, but it would be awesome to have something like that to be builtin generated by vue-cli.

The idea of the component store could be based on npm package names (something like @vue/cli-component), listing cli plugins that install components.

I'm not sure if those should use the very same API as plugins, but just wanted to open this for discussion

**Comment by trickstival** [comment]
I for sure could help with the implementation of something like that if you guys like the idea.

Are there any docs that help the understanding of the project directories / packages structure?
